scene/gui/control.cpp

```diff
@@ -1336,8 +1338,8 @@ void Control::_size_changed() {
 
 	// We use a little workaround to avoid flickering when moving the pivot with _edit_set_pivot()
 	if (is_inside_tree() && Math::abs(Math::sin(data.rotation * 4.0f)) < 0.00001f && get_viewport()->is_snap_controls_to_pixels_enabled()) {
```

Comment:
You should move the test to surround you changes in _update_canvas_item_transform.
And remove its content completely from there, and it does not serve any purpose now.